ehrql/tables/beta/tpp.py

```diff
@@ -42,23 +42,104 @@
 
 @table
 class addresses(EventFrame):
-    address_id = Series(int)
-    start_date = Series(datetime.date)
-    end_date = Series(datetime.date)
-    address_type = Series(int)
-    rural_urban_classification = Series(int)
-    imd_rounded = Series(int, constraints=[Constraint.ClosedRange(0, 32_800, 100)])
+    """
+    Addresses
+
+    Geographic characteristics of the home address a patient registers with a practice.
+    Each row in this table is one registration period per patient.
+    Occasionally, a patient has multiple active registrations on a given date.
+    The postcode from the address is mapped to an Output Area,
+    from which other larger geographic representations can be derived
+    (see various [ONS publications][addresses_ukgeographies] for more detail).
+
+    !!! tip
+        Avoid extracting the rounded IMD ranking to a binary format, such as `.feather` or `.dta`.
+        Either nest it within a variable, such as when grouping rounded IMD by quintile,
+        or extract it to a non-binary format, such as .csv.gz.
+
+        To group rounded IMD ranks by quintile:
+
+        ```py
+        imd = addresses.for_patient_on("2023-01-01").imd_rounded
+        dataset.imd_quintile = case(
+            when((imd >=0) & (imd < int(32844 * 1 / 5))).then("1 (most deprived)"),
+            when(imd < int(32844 * 2 / 5)).then("2"),
+            when(imd < int(32844 * 3 / 5)).then("3"),
+            when(imd < int(32844 * 4 / 5)).then("4"),
+            when(imd < int(32844 * 5 / 5)).then("5 (least deprived)"),
+            default="unknown"
+        )
+        ```
+
+    [addresses_ukgeographies]: https://www.ons.gov.uk/methodology/geography/ukgeographies
+    """
+
+    address_id = Series(
+        int,
+        description="Unique address identifier.",
+    )
+    start_date = Series(
+        datetime.date,
+        description="Start date of registration period.",
```

Comment:
I don't think it's correct to say "registration period" here.  What about "Date patient moved to address"?  And same for `end_date` below.

Reply:
yes, I think that's better